### PR TITLE
openqa-maintenance: Make sure we don't go through requests twice

### LINKED ISSUE #1561: openSUSE maintenance openQA bot adds excessive comments for multi-target incidents
Incidents in openSUSE:Maintenance that have two targets (Leap 15.0 and 42.3) receive excessive comments from the scheduling bot.

e.g.
Visit https://build.opensuse.org/project/show/openSUSE:Maintenance:8160
Fiorentina Linderoth wrote in project openSUSE:Maintenance:8160:
Group openSUSE Leap 42.3 Incidents@DVD-Incidents (3 tests passed)

Visit https://build.opensuse.org/project/show/openSUSE:Maintenance:8160
Fiorentina Linderoth wrote in project openSUSE:Maintenance:8160:
Group openSUSE Leap 15.0 Incidents@DVD-Incidents (3 tests passed)

The comments overwrite each other, also it's a bit of an e-mail spam issue.

### PR DESCRIPTION
**Pull request by coolo**
Fixes #1561

### CODE REVIEW

File: oqamaint/openqabot.py [hunk 1]
```diff
@@ -496,16 +496,18 @@ def incident_openqa_jobs(self, s):
                 'latest': '1'
             })['jobs']
 
-    # for SUSE we use mesh for openSUSE we limit the jobs to open release requests
+    # for SUSE we use mesh, for openSUSE we limit the jobs to open release requests
     def check_opensuse_incidents(self):
-        requests = []
+        requests = dict() # collecting unique requests
```
Comment (mimi1vx) [inline oqamaint/openqabot.py]: dict? I think better is use of `set`

File: oqamaint/openqabot.py [hunk 2]
```diff
@@ -496,16 +496,18 @@ def incident_openqa_jobs(self, s):
...
         self.wait_for_build = set()
         for prj in self.tgt_repo[self.openqa.baseurl].keys():
-            requests += self.ids_project(prj, 'maintenance_release')
+            for r in self.ids_project(prj, 'maintenance_release'):
+                requests[r.reqid] = r
 
         # to be stored in settings
         issues = dict()
-        for req in requests:
+        for req in sorted(requests.keys()):
```
Comment (mimi1vx) [inline oqamaint/openqabot.py]: why sorted ?

File: oqamaint/openqabot.py (hunk 1 above)
Comment (coolo) [reply oqamaint/openqabot.py]: the request object is not hashable

File: oqamaint/openqabot.py (hunk 2 above)
Comment (coolo) [reply oqamaint/openqabot.py]: to make it easier for me to follow the debug output - no other reason

**Pull request merged**